docs/specifications/tests/Zone-TP/zone09.md

```diff
@@ -1,47 +1,296 @@
+[DNS Response] in the same specification.
+
+1.  Create a [DNS Query] with query typ  SOA and query name *Child Zone*
+    ("SOA Query").
+
+2.  Create a [DNS Query] with query typ  MX and query name *Child Zone*
+    ("MX Query").
+
+3.  Obtain the set of name server IP addresses using [Method4] and [Method5]
+    ("Name Server IP").
 
-### Intercase dependencies
+4. Create the following empty sets
+
+    1.  Name server IP address ("No Response MX Query").
+    2.  Name server IP address and associated RCODE value
+        ("Unexpected RCODE MX Response").
+    3.  Name server IP address ("Non-authoritative MX").
+    4.  Name server IP address ("No MX RRset").
+    5.  Name server IP address and associated MX RRset ("MX RRset").
+
+5.  For each name server IP in *Name Server IP* do:
+
+    1. Send *SOA Query* over UDP to the name server.
+    2. Go to next name server IP if at least one of the following criteria is
+       met:
+       1. There is no DNS response.
+       2. The RCODE of the response is not "NoError" ([IANA RCODE List]).
+       3. The AA flag is not set in the response.
+       4. There is no SOA record with owner name matching the query.
+
+    2. Send *MX Query* over UDP to the name server and collect the
+       response, and:
+       1. If the response has the TC flag set, re-query over TCP and use that
+          response instead.
+       2. If there is no DNS response, then add the name server IP to the
+          *No Response MX Query* set.
+       3. Else, if the RCODE of response is not "NoError" ([IANA RCODE List]),
+          then add the name server IP and the RCODE to the
+          *Unexpected RCODE MX Response* set.
+       4. Else, if the AA flag is not set in the response, then add the name
+          server IP to the *Non-authoritative MX* set.
+       5. Else, if there is no MX record with matching owner name in the answer
+          section, then add the name server (IP) to the *No MX RRset* set.
+       6. Else do:
+          1. Extract the MX RRset from the response.
+          2. Add the name server IP and the MX RRset to the *MX RRset* set.
+
+6.  If the set *No Response MX Query* is non-empty, then output
+    *[Z09_NO_RESPONSE_MX_QUERY]* with the name server IP addresses from the set.
+
+7.  If the set *Unexpected RCODE MX Response* is non-empty, then for each RCODE
+    in the set, do:
+    * Output *[Z09_UNEXPECTED_RCODE_MX]* with the RCODE value
+      ([IANA RCODE List]) and the name server IP addresses from the set.
+
+8.  If the set *Non-authoritative MX* is non-empty, then output
+    *[Z09_NON_AUTH_MX_RESPONSE]* with the name server IP addresses from
+    the set.
+
+9.  If both *No MX RRset* set and *MX RRset* set are non-empty then:
+    1. Output *[Z09_INCONSISTENT_MX_RRSETS]*.
+    2. Output *[Z09_NO_MX_FOUND]* with the name server IP addresses from the
+       *No MX RRset* set.
+    3. Output *[Z09_MX_FOUND]* with the name server IP addresses from the
+       *MX RRset* set.
```

Comment:
I am not sure what you mean. I think the MsgID "The name servers return different MX RRsets" will work in both cases, especially since there will always be complementary message tags, Z09_NO_MX_FOUND+Z09_MX_FOUND in the firs case and in the other case multiple Z09_MX_DATA. (I saw a missing plural s on "target" on line 199.)

Reply:
It is a good point. I will update.